yfinance/base.py

```diff
@@ -916,7 +927,7 @@ def get_earnings_dates(self, proxy=None) -> Optional[pd.DataFrame]:
 
         dates = dates.set_index("Earnings Date")
 
-        self._earnings_dates = dates
+        self._earnings_dates[limit] = dates
 
         return dates
 
```

Comment:
Do me favour and delete `get_earnings_history()` below and related API points. It's entirely redundant, an incomplete implementation of `get_earnings_dates()`.

Reply:
Sounds good, removed it.